src/QuickMarcEditor/QuickMarcEditorRows/ControlNumberField/ControlNumberField.js

```diff
@@ -39,45 +37,59 @@ const ControlNumberField = ({
   marcType,
   action,
   recordRows,
+  onChangeControlNumberRecord,
 }) => {
   const intl = useIntl();
-  const { input } = useField(name);
   const { getAuthorityFileNextHrid, isLoading: isLoadingHrid } = useAuthorityFileNextHrid();
 
-  const handleChangeContent = input.onChange;
+  const [selectedSource, setSelectedSource] = useState('');
+
   const contentOf010Row = recordRows.find(row => row.tag === '010')?.content;
   const valueOf010$a = getContentSubfieldValue(contentOf010Row).$a?.[0];
 
-  const [selectedSource, setSelectedSource] = useState('');
+  const controlNumberIndex = recordRows.findIndex(row => row.tag === '001');
+
+  const handleChangeControlNumberRecord = useCallback((content, sourceFile) => {
+    const controlNumberRow = recordRows[controlNumberIndex];
+
+    onChangeControlNumberRecord({
+      index: controlNumberIndex,
+      field: {
+        ...controlNumberRow,
+        content,
+      },
+      sourceFile: sourceFile || controlNumberRow._sourceFile,
+    });
+  }, [onChangeControlNumberRecord, controlNumberIndex, recordRows]);
 
   const handleSourceFileSelection = useCallback(async (sourceFile) => {
     const {
       id: sourceFileId,
       source,
     } = sourceFile;
 
-    let valueOf001Row;
+    let content;
 
     if (source === SOURCES.LOCAL) {
       const { hrid } = await getAuthorityFileNextHrid(sourceFileId);
 
-      valueOf001Row = hrid;
+      content = hrid;
     } else if (source === SOURCES.FOLIO) {
-      valueOf001Row = valueOf010$a;
+      content = valueOf010$a;
     }
 
     setSelectedSource(source);
-
-    handleChangeContent(valueOf001Row);
-  }, [valueOf010$a, handleChangeContent, getAuthorityFileNextHrid]);
+    handleChangeControlNumberRecord(content, sourceFile);
+  }, [valueOf010$a, getAuthorityFileNextHrid, handleChangeControlNumberRecord]);
 
   const canSelectSourceFile = marcType === MARC_TYPES.AUTHORITY && action === QUICK_MARC_ACTIONS.CREATE;
 
   useEffect(() => {
     if (selectedSource === SOURCES.FOLIO) {
-      handleChangeContent(valueOf010$a);
+      handleChangeControlNumberRecord(valueOf010$a);
     }
-  }, [selectedSource, valueOf010$a, handleChangeContent]);
+    /* eslint-disable react-hooks/exhaustive-deps */
+  }, [valueOf010$a]);

```

Comment:
Please add a comment to explain why you know it is safe to ignore `handleChangeControlNumberRecord` here, especially given the previous function included all deps. `exhaustive-deps` is usually pretty reliable.